src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -74,6 +92,40 @@ export const TemplatesCatalogDrawerCreateForm: React.FC<TemplatesCatalogDrawerCr
         });
     };
 
+    const onCustomizeVM = () => {
+      setIsPreparingToCoustomizeVM(true);
+      setCustomizeVMError(undefined);
+
+      k8sCreate<V1Template>({
+        model: ProcessedTemplatesModel,
+        data: template,
+        queryParams: {
+          dryRun: 'All',
+        },
+      }).then((processedTemplate) => {
+        const vmObj = getTemplateVirtualMachineObject(processedTemplate);
+        const updatedVM = produce(vmObj, (vmDraft) => {
+          ensurePath(vmDraft, 'metadata.labels');
+          vmDraft.metadata.name = vmName;
+          vmDraft.metadata.namespace = ns || DEFAULT_NAMESPACE;
+          vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAME] = processedTemplate.metadata.name;
+          vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAMESPACE] =
+            processedTemplate.metadata.namespace;
+        });
+        updateVM(updatedVM)
+          .then(() => {
+            setIsPreparingToCoustomizeVM(false);
+            history.push(
+              `templatescatalog/customize?name=${template.metadata.name}&namespace=${template.metadata.namespace}`,
+            );
+          })
+          .catch((err) => {
+            setIsPreparingToCoustomizeVM(false);
+            setCustomizeVMError(err);
+          });
+      });
+    };
+
```

Comment:
thinking more about this approach.
Instead of injecting the `useWizardVMContext` can't we just add an URL param like vmName or something like that to pass the generated VM name to the customizeForm?

Reply:
I was discussing this with other folks and we came up with using `useWizardVMContext` as the only way.